### PR TITLE
Implement trait constants

### PR DESCRIPTION
**Pull request by fwyzard**
Implement the constpextr template variables
  - `alpaka::isAccelerator<T>`
  - `alpaka::isDevice<T>`
  - `alpaka::isPlatform<T`>
  - `alpaka::isQueue<T>`

that evaluate to `true` if the given type implements the corresponding concept.

### CODE REVIEW

**Comment by fwyzard** [comment]
I've been using something similar to simplify the SFINAE checks in our code, I though they could be useful in general.

**Comment by fwyzard** [comment]
Could anyone have a look at the XCode runners ?
I'm getting the error
```
CMake Error at cmake/alpakaCommon.cmake:391 (message):
-- Could NOT find OpenMP_CXX (missing: OpenMP_CXX_FLAGS OpenMP_CXX_LIB_NAMES) 
  Optional alpaka dependency OpenMP could not be found!
-- Could NOT find OpenMP (missing: OpenMP_CXX_FOUND) 
Call Stack (most recent call first):
-- Configuring incomplete, errors occurred!
  CMakeLists.txt:88 (include)
```
which I think is unrelated to these changes.

**Comment by j-stephan** [comment]
Hm, not sure what is going on there. According to the logs OpenMP is installed correctly. I'll look into it after ACAT if no one else picks it up until then.

**Comment by psychocoderHPC** [comment]
I removed MACOSX from the required passing tests because nobody of us owns an MAC to test whats going on: see https://github.com/alpaka-group/alpaka/issues/1821

I restarted the failing windows CI job.

**Comment by fwyzard** [comment]
Rebased, to re-trigger the tests

**Comment by fwyzard** [comment]
Rebased on top of #1890 (merged).

File: include/alpaka/acc/Traits.hpp [hunk 1]
```diff
@@ -29,6 +29,20 @@ namespace alpaka
     struct ConceptAcc
     {
     };
+
+    // TODO: Remove the following pragmas once support for clang 6 is removed.
+    // They are necessary because clang 6 incorrectly warns about a missing 'extern'.
+#if BOOST_COMP_CLANG
+#    pragma clang diagnostic push
+#    pragma clang diagnostic ignored "-Wmissing-variable-declarations"
+#endif
```
Comment (j-stephan) [inline include/alpaka/acc/Traits.hpp]: These can be removed now, clang-6 is no longer in our support matrix.

**Review by j-stephan — CHANGES_REQUESTED** [review]

**Review by j-stephan — DISMISSED** [review]

**Comment by j-stephan** [comment]
LGTM. Please rebase so the macOS CI update is pulled in. Just restarting the GitHub runners unfortunately didn't do the trick.

**Comment by j-stephan** [comment]
Please rebase (again)... the CI should work again now. Sorry for the noise.

**Comment by fwyzard** [comment]
eventually the CIs will pass...

**Review by j-stephan — APPROVED** [review]
Thanks!

**Pull request merged**